_source/_posts/2023-01-24-jakartaee-auth0.md

```diff
@@ -0,0 +1,545 @@
+
+## Create an Auth0 API and OIDC application
+
+If you have not already, install the [Auth0 CLI](https://github.com/auth0/auth0-cli) and run `auth0 login` in a terminal.
+
+```bash
+Waiting for the login to complete in the browser... done
+
+ ▸    Successfully logged in.
+ ▸    Tenant: dev-0xb84jzp.us.auth0.com
+```
+
+Take note of the domain listed as the tenet. This is your Auth0 domain. If you need to find it again later, you can use `auth0 tenants list`.
+
+You need to create an API on Auth0. The Auth0 API is what exposes identity functionality for all authentication and authorization protocols, such as OpenID Connect and OAuth. **Without the API, Auth0 servers will return an opaque token that the Java application will not be able to verify.** This can lead to some cryptic error messages and some lost time. 
+
+Use the following command to create a custom Auth0 API named `myapi` with the identifier `http://my-api`.
+
+```bash
+auth0 apis create -n myapi --identifier http://my-api
+```
+
+Just press enter three times to accept the default values for scopes, token lifetime, and to allow offline access. The scopes here refer to custom scopes, not the standard scopes (email, profile, and openid) that you will need for OIDC and OAuth.
+
+Now use the Auth0 CLI to create an OpenID Connect (OIDC) application.  From the project base directory, run the following.
+
+```bash
+auth0 apps create
+```
+
+Use the following values:
+
+- **Name**: `javartaee-demo`
+
+- **Description**: whatever you like, or leave blank
+- **Type**: `Regular Web Application`
+- **Callback URLs**: `http://localhost:8080/callback`
+- **Allowed Logout URLs**: `http://localhost:8080`
+
+The console output shows you the Auth0 domain and the OIDC client ID. However, you also need the client secret, which you have to get by logging into Auth0. Type the following:
+
+```bash
+auth0 apps open
+```
+
+Select the OIDC app (or client) you just created from the list. This will open the OIDC application on the Auth0 dashboard.
+
+{% img blog/jakartaee-auth0/oidc-application-auth0.png alt:"Auth0 OIDC App" width:"800" %}{: .center-image }
+
+Fill in the three values in `src/main/resources/openid.properties`. Replace the bracketed values with the values from the OIDC application page on the Auth0 dashboard.
+
+```properties
+issuerUri=<your-auth0-domain>
+clientId=<your-client-id>
+clientSecret=<your-client-secret>ca
+```
+
+## Configure Roles on Auth0
+
+Open your [Auth0 developer dashboard](https://manage.auth0.com). You need to create a role, assign your user to that role, and create an action that will inject the roles into a custom claim in the JWT.
```

Comment:
Is this possible from the CLI? I'm not sure as I haven't looked into it. It certainly would be cool if it was! Can you please check and see if 1.0 has it?